File: integration-test/java/jee/src/main/java/com/qualityminds/lazyval/integration/domain/OrderRepository.java
package com.qualityminds.lazyval.integration.domain;

import com.qualityminds.lazyval.integration.shared.Isbn;

import java.util.List;
import java.util.UUID;

public interface OrderRepository {

    void save(Order order);

    List<Order> findAll();

    Order getById(UUID id);

    List<Order> findByIsbn(Isbn isbn);
}

File: integration-test/kotlin/jee/src/main/kotlin/com/qualityminds/lazyval/integration/domain/OrderRepository.kt
package com.qualityminds.lazyval.integration.domain

import com.qualityminds.lazyval.integration.shared.Isbn
import java.util.UUID

interface OrderRepository {
    fun save(order: Order)
    fun findAll(): List<Order>
    fun getById(id: UUID): Order?
    fun findByIsbn(isbn: Isbn): List<Order>
}
